src/btf_parser.h

```diff
@@ -3,13 +3,171 @@
+
+typedef struct btf_kind_union {
+    std::string name;
+    std::vector<btf_kind_union_member> members;
+    uint32_t size_in_bytes;
+} btf_kind_union;
+
+typedef struct btf_kind_enum_member {
+    std::string name;
+    uint32_t value;
+} btf_kind_enum_member;
+
+typedef struct btf_kind_enum {
+    std::string name;
+    bool is_signed;
+    std::vector<btf_kind_enum_member> members;
+    uint32_t size_in_bytes;
+} btf_kind_enum;
+
+typedef struct btf_kind_fwd {
+    std::string name;
+    bool is_struct;
+} btf_kind_fwd;
+
+typedef struct btf_kind_typedef {
+    std::string name;
+    btf_type_id type;
+} btf_kind_typedef;
+
+typedef struct btf_kind_volatile {
+    btf_type_id type;
+} btf_kind_volatile;
+
+typedef struct btf_kind_const {
+    btf_type_id type;
+} btf_kind_const;
+
+typedef struct btf_kind_restrict {
+    btf_type_id type;
+} btf_kind_restrict;
+
+typedef struct btf_kind_func {
+    std::string name;
+    enum linkage {
+        BTF_LINKAGE_GLOBAL,
+        BTF_LINKAGE_STATIC,
+        BTF_LINKAGE_EXTERN,
+    } linkage;
+    btf_type_id type;
+} btf_kind_func;
+
+typedef struct btf_kind_function_parameter {
+    std::string name;
+    btf_type_id type;
+} btf_kind_function_parameter;
+
+typedef struct btf_kind_function_prototype {
+    std::vector<btf_kind_function_parameter> parameters;
+    btf_type_id return_type;
+} btf_kind_function;
+
+typedef struct btf_kind_var {
+    std::string name;
+    btf_type_id type;
+    enum linkage {
+        BTF_LINKAGE_GLOBAL,
+        BTF_LINKAGE_STATIC,
+    } linkage;
+} btf_kind_var;
+
+typedef struct btf_kind_data_member {
+    btf_type_id type;
+    uint32_t offset;
+    uint32_t size;
+} btf_kind_data_member;
+
+typedef struct btf_kind_data_section {
+    std::string name;
+    std::vector<btf_kind_data_member> members;
+    uint32_t size;
+} btf_kind_data_section;
+
+typedef struct btf_kind_float {
+    std::string name;
+    uint32_t size_in_bytes;
+} btf_kind_float;
+
+typedef struct btf_kind_decl_tag {
+    std::string name;
+    btf_type_id type;
+    uint32_t component_index;
+} btf_kind_decl_tag;
+
+typedef struct btf_kind_type_tag {
+    std::string name;
+    btf_type_id type;
+} btf_kind_type_tag;
+
+typedef struct btf_kind_enum64_member {
+    std::string name;
+    uint64_t value;
+} btf_kind_enum64_member;
+
+typedef struct btf_kind_enum64 {
+    std::string name;
+    bool is_signed;
+    std::vector<btf_kind_enum64_member> members;
+    uint32_t size_in_bytes;
+} btf_kind_enum64;
+
+typedef struct btf_kind_null {
+
+} btf_kind_null;
+
+typedef std::variant<btf_kind_null, btf_kind_int, btf_kind_ptr, btf_kind_array, btf_kind_struct, btf_kind_union,
+                     btf_kind_enum, btf_kind_fwd, btf_kind_typedef, btf_kind_volatile, btf_kind_const,
+                     btf_kind_restrict, btf_kind_func, btf_kind_function, btf_kind_var, btf_kind_data_section,
+                     btf_kind_float, btf_kind_decl_tag, btf_kind_type_tag, btf_kind_enum64>
+    btf_kind;
```

Comment:
```suggestion
using btf_kind = std::variant<btf_kind_null, btf_kind_int, btf_kind_ptr, btf_kind_array, btf_kind_struct, btf_kind_union,
                     btf_kind_enum, btf_kind_fwd, btf_kind_typedef, btf_kind_volatile, btf_kind_const,
                     btf_kind_restrict, btf_kind_func, btf_kind_function, btf_kind_var, btf_kind_data_section,
                     btf_kind_float, btf_kind_decl_tag, btf_kind_type_tag, btf_kind_enum64>;
```

Reply:
Similarly everywhere else. Use typedef when it's either syntactically inconvenient to use `using` (e.g. function pointers) or when you need C compatibility.